packages/@jsii/go-runtime/jsii-experimental/runtime.go

```diff
@@ -0,0 +1,52 @@
+package jsii
+
+import (
+	"sync"
+)
+
+var (
+	clientInstance      *client
+	clientInstanceMutex sync.Mutex
+	once                sync.Once
+)
+
+// getClient returns a singleton client instance, initializing one the first
+// time it is called.
```

Comment:
I assume that this was pulled out of https://github.com/aws/jsii/pull/2046/files#diff-cc08ae342c04a4072f9192cab59bf8e7R12, but wondering now if there any reason not to put all this code in the `client.go` file? Usually file names correspond to custom type names (in the previous PR, `runtime`) so it's a little confusing for `getClient` to float around in here now.

Reply:
This actually crossed my mind and I guess you're right. I didn't see the value of the `runtime` type since it literally only wraps a single client instance... But the `client.go` file could actually manage the singleton itself, and it wouldn't change much of the rest.